src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -1215,7 +1200,7 @@ private String getNearestClassFrameName() {
      */
     private static boolean isLambdaParameter(DetailAST ast) {
         DetailAST parent;
-        for (parent = ast.getParent(); parent != null; parent = parent.getParent()) {
```

Comment:
Similar to https://github.com/checkstyle/checkstyle/pull/12628#discussion_r1068204516 .